api/src/test/java/org/openmrs/api/impl/PatientServiceImplUnitTest.java

```diff
@@ -0,0 +1,1379 @@
+
+		when(administrationService.getGlobalProperty("concept.reasonExitedCare")).thenReturn(property);
+		when(conceptService.getConcept(property)).thenReturn(concept);
+		when(obsService.getObservationsByPersonAndConcept(patient, concept)).thenReturn(observations);
+		when(concept.getName()).thenReturn(conceptName);
+		when(obsService.saveObs(observation1, "updating")).thenReturn(observation1);
+
+		patientServiceImpl.exitFromCare(patient, newDate, concept);
+
+		verify(administrationService).getGlobalProperty("concept.reasonExitedCare");
+		verify(conceptService).getConcept(property);
+		verify(obsService).getObservationsByPersonAndConcept(patient, concept);
+		verify(concept, times(1)).getName();
+		verify(obsService).saveObs(observation1, "updated by PatientService.saveReasonForExit");
+	}
+
+	@Test
+	public void exitFromCare_whenHasAMoreThanOneReasonToExit_shouldNotSaveObservation() {
+
+		Obs observation2 = new Obs(2);
+		observations.add(observation2);
+
+		when(administrationService.getGlobalProperty("concept.reasonExitedCare")).thenReturn(property);
+		when(conceptService.getConcept(property)).thenReturn(concept);
+		when(obsService.getObservationsByPersonAndConcept(patient, concept)).thenReturn(observations);
+
+		patientServiceImpl.exitFromCare(patient, newDate, concept);
+
+		verify(administrationService).getGlobalProperty("concept.reasonExitedCare");
+		verify(conceptService).getConcept(property);
+		verify(obsService).getObservationsByPersonAndConcept(patient, concept);
+		verify(concept, never()).getName();
+		verify(obsService, never()).saveObs(any(Obs.class), eq("updated by PatientService.saveReasonForExit"));
+
+	}
+
+	@Test
+	public void exitFromCare_whenHasNoObservation_shouldCreateAnObservationAndSaveIt() {
+
+		when(administrationService.getGlobalProperty("concept.reasonExitedCare")).thenReturn(property);
+		when(conceptService.getConcept(property)).thenReturn(concept);
+		when(obsService.getObservationsByPersonAndConcept(patient, concept)).thenReturn(observations);
+		when(obsService.saveObs(observation1, "updating")).thenReturn(observation1);
+		when(locationService.getDefaultLocation()).thenReturn(location);
+		when(concept.getName()).thenReturn(conceptName);
+		when(obsService.getObservationsByPersonAndConcept(patient, concept)).thenReturn(new ArrayList<>());
+
+		patientServiceImpl.exitFromCare(patient, newDate, concept);
+
+		verify(administrationService).getGlobalProperty("concept.reasonExitedCare");
+		verify(conceptService).getConcept(property);
+		verify(obsService).getObservationsByPersonAndConcept(patient, concept);
+		verify(locationService).getDefaultLocation();
+		verify(concept, times(1)).getName();
+		verify(obsService).saveObs(any(Obs.class), eq("updated by PatientService.saveReasonForExit"));
+	}
+
+	@Test
+	public void exitFromCare_whenHasNoObservationAndNoDefaultLocation_shouldCreateAnObservationWithoutLocationAndSaveIt() {
```

Comment:
@ayush2297 you can do something like this `shouldExitFromCareWhenNoObservationAndDefaultLocation` as advised here https://wiki.openmrs.org/display/docs/Unit+Testing+Conventions to maintain uniformity

Reply:
I am quite confused about this one, I used to follow a completely different way for test name : "given..._when..._then..." , but here i saw many tests following one convention : "methodName_should...." so i tried to follow that ...........

can you confirm that? should test names be starting from "should...." or "methodName_shoud..." ?

Reply:
@gitcliff i have finished other changes.... only this remains.. can you confirm which approach should i follow?

Reply:
@ayush2297 I would say that since your tests are going to go into PatientServiceImplTest and in there we use `methodName_shouldYYY` stick to that naming convention.